src/hotspot/share/oops/oop.inline.hpp

```diff
@@ -213,6 +213,9 @@ bool oopDesc::is_typeArray() const { return klass()->is_typeArray_klass(); }
 
 void*    oopDesc::field_addr(int offset)     const { return reinterpret_cast<void*>(cast_from_oop<intptr_t>(as_oop()) + offset); }
 
+template<typename T>
+T*       oopDesc::field_addr(int offset)     const { return reinterpret_cast<T*>(cast_from_oop<intptr_t>(as_oop()) + offset); }
+
 template <class T>
 T*       oopDesc::obj_field_addr(int offset) const { return (T*) field_addr(offset); }
```

Comment:
The `template<typename T>` doesn't match the style used for obj_field_addr.

Could/should we consolidate the three functions?:
```
void*    oopDesc::field_addr(int offset)     const { return reinterpret_cast<void*>(cast_from_oop<intptr_t>(as_oop()) + offset); }

template <typename T>
T*       oopDesc::field_addr(int offset)     const { return reinterpret_cast<T*>(field_addr(offset)); }

template <typename T>
T*       oopDesc::obj_field_addr(int offset) const { return field_addr<T>(offset); }
```